tests/solver-by-substitution.cc

```diff
@@ -841,6 +841,8 @@ BOOST_AUTO_TEST_CASE (rightHandSide)
     vector_t x (N);
     solver.solve (x);
```

Comment:
Adding `TARGET_COMPILE_OPTIONS(solver-by-substitution PRIVATE -DEIGEN_INITIALIZE_MATRICES_BY_NAN)` to `tests/CMakeLists.txt` pointed out the issue:

@florent-lamiraux  vector `x` was not initialized.